### PR TITLE
[BUGFIX] Check for datetime-parseable strings in validate_metric_value_between_configuration

### PR DESCRIPTION
**Pull request by victorwyee**
Changes proposed in this pull request:
- Refactors `ExpectColumnValuesToBeBetween` to use config validation code from grandparent `TableExpectation` class.
- Adds two unit tests for `ExpectColumnValuesToBeBetween` with `parse_strings_as_datetimes=true`.
- Attempts to reproduce #2341

### CODE REVIEW

**Comment by victorwyee** [comment]
Well, `pytest tests/test_definitions/test_expectations.py` passed fine, but upon inspection of the build failures here, I see that running `pytest --no-postgresql --no-spark --no-sqlalchemy` also produces the same errors

Namely

```
>           raise InvalidExpectationConfigurationError(str(e))
E           great_expectations.exceptions.exceptions.InvalidExpectationConfigurationError: Provided min threshold must be a number

/opt/projects/great_expectations/great_expectations/expectations/expectation.py:1106: InvalidExpectationConfigurationError
```

and

```
>           raise InvalidExpectationConfigurationError(str(e))
E           great_expectations.exceptions.exceptions.InvalidExpectationConfigurationError: min_value and max_value cannot both be None
```

File: great_expectations/expectations/expectation.py [hunk 1]
```diff
@@ -1081,6 +1081,9 @@ def validate_metric_value_between_configuration(
 
         try:
             # Ensuring Proper interval has been provided
+            assert (
```
Comment (victorwyee) [inline great_expectations/expectations/expectation.py]: I wasn't expecting this to break a bunch of stuff. This seems like a reasonable assertion.

File: great_expectations/util.py [hunk 2]
```diff
@@ -889,6 +891,14 @@ def is_float(value: Any) -> bool:
     return True
 
 
+def is_parseable_date(value: Any, fuzzy: bool=False) -> bool:
+    try:
+        parsed_date: datetime = parse(value, fuzzy=fuzzy)
```
Comment (victorwyee) [inline great_expectations/util.py:913]: I'm not sure if how well strings that `datetime.parse` can parse will align with the various engines that the datetime string will be fit into. But this seems to work for pandas.

**Comment by victorwyee** [comment]
`pytest --no-postgresql --no-spark --no-sqlalchemy tests/test_definitions/test_expectations_cfe.py` passes locally, so I suspect at least one engine is returning dates in a different format and/or causing the comparison(s) to fail. Planning on installing the other backends soon to try to reproduce locally and get a fix in.

**Comment by spbail** [comment]
@victorwyee Howdy! Thanks for the contribution! Mind marking this as draft while you're still working on it? Also, feel free to ping us in Slack if you'd like some help with the failing tests. (We're doing an all-hands this week so might be a little less responsive though.)

**Comment by victorwyee** [comment]
Was able to repro the failing test after installing Spark locally and running `pytest tests/test_definitions/test_expectations_cfe.py`. Tests pass with pandas, postgres, and sqlalchemy/mysql backends.

**Comment by Shinnnyshinshin** [comment]
HI @victorwyee. First, thank you for putting together this elegant PR. Your solution that uses `datetime.parse` seems to work well for all the backends that GE currently supports. There is only test that currently fails for the `spark` backend, and it can (AFIK) be addressed by adding the `parse_strings_as_datetimes` logic to to the `_spark` fixture in the `column_values_between.py` metric. 

Let me know the following works for you 

```
@column_condition_partial(engine=SparkDFExecutionEngine)
    def _spark(
        cls,
        column,
        min_value=None,
        max_value=None,
        strict_min=None,
        strict_max=None,
        parse_strings_as_datetimes=None,
        **kwargs
    ):
        if parse_strings_as_datetimes:
            if min_value:
                min_value = parse(min_value)
            if max_value:
                max_value = parse(max_value)
```

**Review by Shinnnyshinshin — CHANGES_REQUESTED** [review]
Once we get the test passing for spark I think we should be good to go

**Comment by victorwyee** [comment]
> HI @victorwyee. First, thank you for putting together this elegant PR. Your solution that uses `datetime.parse` seems to work well for all the backends that GE currently supports. There is only test that currently fails for the `spark` backend, and it can (AFIK) be addressed by adding the `parse_strings_as_datetimes` logic to to the `_spark` fixture in the `column_values_between.py` metric.
> 
> Let me know the following works for you
> 
> ```
> @column_condition_partial(engine=SparkDFExecutionEngine)
>     def _spark(
>         cls,
>         column,
>         min_value=None,
>         max_value=None,
>         strict_min=None,
>         strict_max=None,
>         parse_strings_as_datetimes=None,
>         **kwargs
>     ):
>         if parse_strings_as_datetimes:
>             if min_value:
>                 min_value = parse(min_value)
>             if max_value:
>                 max_value = parse(max_value)
> ```

Thanks! I decided to take the `datetime.parse` approach partly because it's already used in many other places in `great_expectations`, and I generally prioritize consistency in a codebase. It helps that it works well for all supported engines too!

On adding support for Spark, the solution is obvious now, but I couldn't find it at first glance. Thanks for the pointer! All tests pass on my local now for all engines. :)

File: great_expectations/expectations/core/expect_column_values_to_be_between.py [hunk 3]
```diff
@@ -125,51 +125,7 @@ def validate_configuration(self, configuration: Optional[ExpectationConfiguratio
...
-        if "min_value" in configuration.kwargs:
-            min_val = configuration.kwargs["min_value"]
-
-        if "max_value" in configuration.kwargs:
-            max_val = configuration.kwargs["max_value"]
-
-        try:
-            # Ensuring Proper interval has been provided
-            assert (
-                min_val is not None or max_val is not None
```
Comment (victorwyee) [inline great_expectations/expectations/core/expect_column_values_to_be_between.py:144]: I couldn't get this assertion to work for `tests/test_definitions/test_expectations_cfe.py`. There're several cases where the test runner thinks that `min_val` and `max_val` are both None.

Comment (Shinnnyshinshin) [reply great_expectations/expectations/core/expect_column_values_to_be_between.py:144]: Hi @victorwyee I'm actually having some trouble re-creating this error on my end. Does the error persist with the following code? (they should be testing the same condition)

`
assert any([min_val, max_val]), "min_value and max_value cannot both be None"
`

Comment (victorwyee) [reply great_expectations/expectations/core/expect_column_values_to_be_between.py:144]: Yes, adding this assertion still fails on my end. Note that this wasn't in the original configuration validation function before -- just in this specific expectation (which I've refactored).

This is what I get

```
=============================================== short test summary info ================================================
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_map_expectations/expect_column_values_to_be_between:missing_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_map_expectations/expect_column_values_to_be_between:null_max_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_stdev_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_stdev_to_be_between:positive_zero_stdev_exact_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_mean_to_be_between:type_mismatch_null_observed_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_stdev_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_stdev_to_be_between:positive_zero_stdev_exact_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_mean_to_be_between:type_mismatch_null_observed_value]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_max_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_mean_to_be_between:simple_mean_includes_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_mean_to_be_between:vacuously_true_missing_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_unique_value_count_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_median_to_be_between:vacuously_true_both_min_and_max_null]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_median_to_be_between:test_empty_column_should_be_false_no_observed_value_with_which_to_compare]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_min_to_be_between:negative_test_case_with_only_a_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/column_aggregate_expectations/expect_column_proportion_of_unique_values_to_be_between:vacuously_true_null_min_and_max]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[pandas/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[spark/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[sqlite/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/other_expectations/expect_table_row_count_to_be_between:vacuously_true]
FAILED tests/test_definitions/test_expectations_cfe.py::test_case_runner_cfe[postgresql/other_expectations/expect_table_column_count_to_be_between:vacuously_true]
========================= 54 failed, 1259 passed, 483 skipped, 5 warnings in 147.71s (0:02:27) =========================
```

Example exception:
```
______ test_case_runner_cfe[postgresql/other_expectations/expect_table_column_count_to_be_between:vacuously_true] ______

self = <great_expectations.expectations.core.expect_table_column_count_to_be_between.ExpectTableColumnCountToBeBetween object at 0x7f89e8e04550>
configuration = {"kwargs": {"result_format": "COMPLETE", "include_config": false}, "meta": {}, "expectation_type": "expect_table_column_count_to_be_between"}

    def validate_metric_value_between_configuration(
        self, configuration: Optional[ExpectationConfiguration]
    ):
        # Validating that Minimum and Maximum values are of the proper format and type
        min_val = None
        max_val = None
        parse_strings_as_datetimes = None

        if "min_value" in configuration.kwargs:
            min_val = configuration.kwargs["min_value"]

        if "max_value" in configuration.kwargs:
            max_val = configuration.kwargs["max_value"]

        if "parse_strings_as_datetimes" in configuration.kwargs:
            parse_strings_as_datetimes = configuration.kwargs[
                "parse_strings_as_datetimes"
            ]

        try:
            # Ensuring Proper interval has been provided
            assert any(
                [min_val, max_val]
>           ), "min_value and max_value cannot both be None"
E           AssertionError: min_value and max_value cannot both be None

great_expectations/expectations/expectation.py:1093: AssertionError

During handling of the above exception, another exception occurred:

test_case = {'expectation_type': 'expect_table_column_count_to_be_between', 'skip': False, 'test': {'exact_match_out': False, 'in'... 'vacuously_true'}, 'validator_with_data': <great_expectations.validator.validator.Validator object at 0x7f89e44833d0>}

    def test_case_runner_cfe(test_case):
        if test_case["skip"]:
            pytest.skip()

        # Note: this should never be done in practice, but we are wiping expectations to reuse batches during testing.
        # test_case["batch"]._initialize_expectations()

        evaluate_json_test_cfe(
            validator=test_case["validator_with_data"],
            expectation_type=test_case["expectation_type"],
>           test=test_case["test"],
        )

tests/test_definitions/test_expectations_cfe.py:299:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_utils.py:1208: in evaluate_json_test_cfe
    result = getattr(validator, expectation_type)(**runtime_kwargs)
great_expectations/validator/validator.py:263: in inst_expectation
    raise err
great_expectations/validator/validator.py:213: in inst_expectation
    expectation = expectation_impl(configuration)
great_expectations/expectations/expectation.py:137: in __init__
    self.validate_configuration(configuration)
great_expectations/expectations/core/expect_table_column_count_to_be_between.py:107: in validate_configuration
    self.validate_metric_value_between_configuration(configuration=configuration)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <great_expectations.expectations.core.expect_table_column_count_to_be_between.ExpectTableColumnCountToBeBetween object at 0x7f89e8e04550>
configuration = {"kwargs": {"result_format": "COMPLETE", "include_config": false}, "meta": {}, "expectation_type": "expect_table_column_count_to_be_between"}

    def validate_metric_value_between_configuration(
        self, configuration: Optional[ExpectationConfiguration]
    ):
        # Validating that Minimum and Maximum values are of the proper format and type
        min_val = None
        max_val = None
        parse_strings_as_datetimes = None

        if "min_value" in configuration.kwargs:
            min_val = configuration.kwargs["min_value"]

        if "max_value" in configuration.kwargs:
            max_val = configuration.kwargs["max_value"]

        if "parse_strings_as_datetimes" in configuration.kwargs:
            parse_strings_as_datetimes = configuration.kwargs[
                "parse_strings_as_datetimes"
            ]

        try:
            # Ensuring Proper interval has been provided
            assert any(
                [min_val, max_val]
            ), "min_value and max_value cannot both be None"
            if parse_strings_as_datetimes:
                assert min_val is None or is_parseable_date(
                    min_val
                ), "Provided min threshold must be a dateutil-parseable date"
                assert (
                    max_val is None or is_parseable_date(max_val),
                ), "Provided max threshold must be a dateutil-parseable date"
            else:
                assert min_val is None or isinstance(
                    min_val, (float, int, dict)
                ), "Provided min threshold must be a number"
                if isinstance(min_val, dict):
                    assert (
                        "$PARAMETER" in min_val
                    ), 'Evaluation Parameter dict for min_value kwarg must have "$PARAMETER" key'

                assert max_val is None or isinstance(
                    max_val, (float, int, dict)
                ), "Provided max threshold must be a number"
                if isinstance(max_val, dict):
                    assert "$PARAMETER" in max_val, (
                        "Evaluation Parameter dict for max_value "
                        "kwarg "
                        'must have "$PARAMETER" key'
                    )

        except AssertionError as e:
>           raise InvalidExpectationConfigurationError(str(e))
E           great_expectations.exceptions.exceptions.InvalidExpectationConfigurationError: min_value and max_value cannot both be None

great_expectations/expectations/expectation.py:1121: InvalidExpectationConfigurationError
```

File: tests/test_definitions/column_map_expectations/expect_column_values_to_be_between.json [hunk 4]
```diff
@@ -178,6 +193,21 @@
           "min_value": 3
         }
       },
+      {
```
Comment (Shinnnyshinshin) [inline tests/test_definitions/column_map_expectations/expect_column_values_to_be_between.json:196]:
```suggestion
     {
        "title" : "null_min_max_value",
        "exact_match_out": false,
        "out": {
          "traceback_substring": "min_value and max_value cannot both be None"
        },
        "in": {
          "column": "x",
          "min_value": null,
          "max_value": null,
          "catch_exceptions": true
        }
      },
     {
```

File: great_expectations/expectations/core/expect_column_values_to_be_between.py [hunk 5]
```diff
@@ -125,51 +125,7 @@ def validate_configuration(self, configuration: Optional[ExpectationConfiguratio
 
         # Setting up a configuration
         super().validate_configuration(configuration)
-        if configuration is None:
-            configuration = self.configuration
-
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_values_to_be_between.py]: `


        if configuration is None:
            configuration = self.configuration

        min_val = None
        max_val = None
        # Setting these values if they are available
        if "min_value" in configuration.kwargs:
            min_val = configuration.kwargs["min_value"]

        if "max_value" in configuration.kwargs:
            max_val = configuration.kwargs["max_value"]
        try:
            # Ensuring Proper interval has been provided
            assert (
                min_val is not None or max_val is not None
            ), "min_value and max_value cannot both be None"
        except AssertionError as e:
            raise InvalidExpectationConfigurationError(str(e))
`

**Comment by Shinnnyshinshin** [comment]
hi @victorwyee I got a chance to do some digging this afternoon and (I think) figured out why the assertion was failing. 

The expectation `ExpectColumnValuesToBeBetween ` behaves a little differently from other expectations that use the `validate_metric_value_between_configuration()` function.  Other expectations of this type include `expect_column_mean_to_be_between` and `expect_column_median_to_be_between`,  and they support a `vacuously_true` condition, which means the expectation will pass if the min/max values set to `null`. These were the tests that were failing when we added the assertion to  `validate_metric_value_between_configuration()`

In the case of this PR I think we can try the following 2 thing to ensure `ExpectColumnValuesToBeBetween` behaves consistently with the existing code.

1) re-add the following assertion to `great_expectations/expectations/core/expect_column_values_to_be_between.py`

```
 assert (
            min_val is not None or max_val is not None
        ), "min_value and max_value cannot both be None"
```

2) adding a test for the situation where both the min_value and max_value are `None` to `expect_column_values_to_be_between.json`

I've added 2 suggestions for the changes.  Could you have a look and let me know what you think?  I was able to get the tests to pass locally.

**Comment by Shinnnyshinshin** [comment]
Hi @victorwyee I just wanted to follow up on this PR. This is a great fix for the issue, and we're excited to have it be part of the codebase. If you're not able to able to get to the remaining suggestions in the next few days (let's say 2/24/21?) would it be ok if I got it over the line?  You've basically done 90% of the work, so it would be a minor tweak on my end.

**Comment by victorwyee** [comment]
Oh definitely planning on closing this out today or tomorrow. Sorry I
didn't get to this at the end of last week! Will work on this in a few
hours 😄

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you very much 🚀 LGTM

**Comment by Shinnnyshinshin** [comment]
@victorwyee I just approved, and will merge once tests pass. Thank you very much for your contribution here, and I hope we get to work again in the future :) have a great one!

**Comment by Shinnnyshinshin** [comment]
A small update : currently our CI pipelines are failing for reasons that are un-related to your PR. We'll merge as soon as it gets cleared up :)

**Comment by victorwyee** [comment]
> A small update : currently our CI pipelines are failing for reasons that are un-related to your PR. We'll merge as soon as it gets cleared up :)

Yeah, I took a look and saw the timeouts earlier. Thanks for the update!
